src/main/java/duke/list/TaskList.java

```diff
@@ -0,0 +1,158 @@
+package duke.list;
+
+import duke.exceptions.IllegalOperation;
+import duke.ui.MessageBubble;
+import duke.task.Task;
+
+import java.util.ArrayList;
+
+public class TaskList {
+    public static int MAX_LIST_ITEMS = 1000;
+    private final ArrayList<Task> items = new ArrayList<>();
+
+    /**
+     * Convenience constructor
+     */
+    public TaskList() {
+    }
+
+    /**
+     * Add the specified Task to the end of the TaskList.
+     * A message will be print out to show the result.
+     *
+     * @param task the task to be added
+     * @throws IllegalOperation if the number of tasks >= MAX_LIST_ITEMS
+     */
+    public void addItem(Task task) throws IllegalOperation {
+        addItem(task, true);
```

Comment:
Is it necessary to add a "true" to this method? Consider extracting it so that the method is cleaner (right now, it is unclear what is effected if "true" is added).